### PR TITLE
fix: Reroute chat page, Rename 'friend-list' component, embed link into logo, and Improve module and name input performance

### PR DESCRIPTION
**Pull request by dipasukhaedbert**

### CODE REVIEW

File: src/components/header/header.component.jsx [hunk 1]
```diff
@@ -61,7 +61,11 @@ const Header = () => {
             <i class="fas fa-bars" id="btn"></i>
             <i class="fas fa-times" id="cancel"></i>
           </label>
-          <h1 href="/quick-match">ModMate</h1>
+          <h1>
+            <Link to="/profile" style={{ color: "white" }}>
+              ModMate
+            </Link>
+          </h1>
```
Comment (felixhalim) [inline src/components/header/header.component.jsx:68]: Try Link wrap h1, no inline css required.

File: src/firebase/index.firebase.js [hunk 2]
```diff
@@ -17,5 +17,6 @@ firebase.initializeApp({
 const db = firebase.firestore();
 const auth = firebase.auth();
 const stor = firebase.storage();
+const realdb = firebase.database();
```
Comment (felixhalim) [inline src/firebase/index.firebase.js:20]: Is it possible to use cloud firestore instead of realtime db?

Comment (felixhalim) [reply src/firebase/index.firebase.js:20]: We have 2 choices here, continue implementing using realtime db which will save our free firestore read limit or change to cloud firestore

File: src/components/header/header.component.jsx (hunk 1 above)
Comment (dipasukhaedbert) [reply src/components/header/header.component.jsx:68]: Noted

File: src/firebase/index.firebase.js (hunk 2 above)
Comment (dipasukhaedbert) [reply src/firebase/index.firebase.js:20]: Yes

Comment (felixhalim) [reply src/firebase/index.firebase.js:20]: Okay, I think it'd be better to implement realtime db, what do you think?

Comment (dipasukhaedbert) [reply src/firebase/index.firebase.js:20]: Yes, it is possible.

**Comment by felixhalim** [comment]
Reached a consensus of using realtimedb as chat database solution.

**Pull request merged**